akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -116,22 +120,29 @@ private[cluster] final class ClusterHeartbeatSender extends Actor with ActorLogg
   }
 
   def init(snapshot: CurrentClusterState): Unit = {
-    val nodes: Set[UniqueAddress] = snapshot.members.map(_.uniqueAddress)
-    val unreachable: Set[UniqueAddress] = snapshot.unreachable.map(_.uniqueAddress)
+    val nodes = snapshot.members.collect({ case m if filterInternalClusterMembers(m) ⇒ m.uniqueAddress })
+    val unreachable = snapshot.unreachable.collect({ case m if filterInternalClusterMembers(m) ⇒ m.uniqueAddress })
     state = state.init(nodes, unreachable)
   }
 
   def addMember(m: Member): Unit =
-    if (m.uniqueAddress != selfUniqueAddress && !state.contains(m.uniqueAddress))
+    if (m.uniqueAddress != selfUniqueAddress && // is not self 
+      !state.contains(m.uniqueAddress) && // not already added
+      filterInternalClusterMembers(m) // should be watching members from this team (internal / external)
+      ) {
+      log.info(s"WILL MONITOR NODE ===== ${m.uniqueAddress} ======== TEAM: ${m.team} (myself team: ${cluster.selfDataCenter})")
       state = state.addMember(m.uniqueAddress)
```

Comment:
When the set of monitored nodes changes the ones that we stopped monitoring must be removed from the failure detector. That can happen both in `addMember` and `removeMember`.